requests_cache/backends/base.py

```diff
@@ -152,6 +143,35 @@ def has_url(self, url: str) -> bool:
         """Returns `True` if cache has `url`, `False` otherwise. Works only for GET request urls"""
         return self.has_key(url_to_key(url, self.ignored_parameters))  # noqa: W601
 
+    def keys(self) -> Iterator[str]:
+        """Get all cache keys for redirects and (valid) responses combined"""
+        yield from self.redirects.keys()
+        for key, _ in self._get_valid_responses():
+            yield key
+
+    def values(self) -> Iterator[CachedResponse]:
+        """Get all valid response objects from the cache"""
+        for _, response in self._get_valid_responses():
+            yield response
+
+    def _get_valid_responses(self, delete_invalid=False) -> Iterator[Tuple[str, CachedResponse]]:
+        """Get all responses from the cache, and skip (+ optionally delete) any invalid ones that
+        can't be deserialized"""
+        keys_to_delete = []
+
+        for key in self.responses.keys():
+            try:
+                yield key, self.responses[key]
+            except Exception as e:
```

Comment:
Perhaps more specific exceptions here? `(KeyError, TypeError)`.

... But they are rather generic too... Perhaos future or then, adding Exception specific to the error.

Reply:
Also `AttributeError`, `PickleError`, maybe a couple more in some weird edge cases for backends other than SQLite. But I agree, I don't really like `except Exception`.